src/test/compile-fail/nll/issue-27282-guards-always-borrow.rs

```diff
@@ -0,0 +1,61 @@
+// Copyright 2012-2014 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//revisions: ast mir
+//[mir] compile-flags: -Z borrowck=mir -Z nll
+
+#![feature(rustc_attrs)]
+
+// Here is arielb1's basic example from rust-lang/rust#27282
+// that AST borrowck is flummoxed by:
+
```

Comment:
Yes, this was not intended to fix *everything*.

Reply:
Just an FYI: I agree that the this did not put in the full fix to the core problem. However, the example outlined by @arielb1 is *not* accepted by the compiler (under NLL) once this PR is in place. In particular, you get these errors:

```
error[E0382]: use of moved value: `*v`
  --> ../../ariel-example-2-27282.rs:11:9
   |
8  |             (|| { let bar = v; bar.take() })();
   |             -------------------------------- value moved here
...
11 |         Some(s) => std::process::exit(*s),
   |         ^^^^^^^ value used here after move

error[E0382]: use of moved value: `v.0`
  --> ../../ariel-example-2-27282.rs:11:14
   |
8  |             (|| { let bar = v; bar.take() })();
   |             -------------------------------- value moved here
...
11 |         Some(s) => std::process::exit(*s),
   |              ^ value used here after move
   |
   = note: move occurs because `v.0` has type `&i32`, which does not implement the `Copy` trait
```